src/go/rpk/pkg/cli/cluster/storage/mount.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package storage
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newMountCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var to string
+
+	cmd := &cobra.Command{
+		Use:   "mount [TOPIC]",
+		Short: "Mount a topic to the Redpanda Cluster from Tiered Storage",
+		Long: `Mount a topic from Tiered Storage to the Redpanda cluster.
+
+This command mounts a topic in the Redpanda Cluster using log segments stored
+in Tiered Storage.
+
+Requirements:
+- Tiered storage must be enabled.
+- The topic must have a minimum of three partitions.
+- Log segments for the topic must be available in Tiered Storage.
+- A topic with the same name must not already exist in the cluster.`,
+		Example: `
+Mounts topic my-typic from Tiered Storage to the cluster in the my-namespace
+	rpk cluster storage mount my-topic
+
+Mount topic my-topic from Tiered Storage to the cluster in the my-namespace 
+with my-new-topic as the new topic name
+	rpk cluster storage mount my-namespace/my-topic --to my-namespace/my-new-topic
+`,
+		Args: cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, from []string) {
+			pf, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			config.CheckExitCloudAdmin(pf)
+			adm, err := adminapi.NewClient(cmd.Context(), fs, pf)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			n, t := nsTopic(from[0])
+			if t == "" {
+				out.Die("topic is required")
+			}
+			topic := rpadmin.InboundTopic{
+				SourceTopic: rpadmin.NamespacedTopic{
+					Namespace: string2pointer(n),
+					Topic:     t,
+				},
+			}
+			an, at := nsTopic(to)
+			if at != "" {
+				topic.Alias = &rpadmin.NamespacedTopic{
+					Namespace: string2pointer(an),
+					Topic:     at,
+				}
+			}
```

Comment:
Just confirming... What's the behavior of `to` being empty? Does it mount to the same `from` topic?

Reply:
Yes